### PR TITLE
[Triton] Apply fused vision rotary pos embedding to Qwen2.5-VL: up to…

### PR DESCRIPTION
**Pull request by flutist**
## Motivation

Following the Triton kernel introduced for Qwen2-VL, this PR applies the same optimization to `Qwen2_5_VisionTransformer.rot_pos_emb`. The method has the identical Python for-loop + `torch.cat` + gather pattern that scales poorly with multi-image batches.

### `python/sglang/srt/models/qwen2_5_vl.py` (+14 lines)

Added early-return fast path at the top of `rot_pos_emb` that calls the existing `triton_vision_rot_pos_emb` kernel on CUDA. Original fallback untouched. No new imports needed (`_is_cuda` already exists at module level).

## Accuracy Tests

```
10 passed, 3 subtests passed ✅ (atol=1e-5, rtol=1e-5)
```

Same kernel, same test suite (`test/registered/rotary/test_vision_rotary_triton.py`).

## Speed Tests

Identical kernel as Qwen2-VL, same benchmark results apply:

| Scenario | Original | Triton | Speedup |
|----------|----------|--------|---------|
| 32 images 720p | 8.96ms | 0.51ms | **17.7x** |
| 16 images 1080p | 9.53ms | 0.56ms | **17.0x** |
| 32 images 1080p | 16.32ms | 1.05ms | **15.6x** |
| 16 mixed 720p~1080p+video | 7.35ms | 0.46ms | **15.9x** |
| 1 image 1080p | 1.48ms | 0.12ms | **12.3x** |

## End-to-End Evaluation (lmms_eval)

Model: Qwen2.5-VL-3B-Instruct, Benchmark: MMMU val (900 samples)

### Accuracy

| Version | MMMU acc | Output tokens |
|---------|----------|---------------|
| Baseline (original) | 0.4722 | 2,634 |
| Triton kernel | 0.4667 | 2,632 |

Difference of 5/900 questions (0.55%) is within normal sampling variance. Numerical precision verified at atol=1e-5.

### Throughput

| Metric | Baseline | Triton kernel |
|--------|----------|---------------|
| Total elapsed time | 1989.86s | 1938.27s |
| Total gen tokens | 2,634 | 2,632 |
| Avg speed | 1.32 tokens/s | 1.36 tokens/s |

### Reproduce

```bash
# Install lmms_eval from source
pip install git+https://github.com/EvolvingLMMs-Lab/lmms-eval.git

# Launch SGLang server
sglang serve --model-path Qwen/Qwen2.5-VL-3B-Instruct --port 30000

# Run evaluation
OPENAI_API_KEY=dummy python3 -m lmms_eval \
    --model openai_compatible \
    --model_args model_version=Qwen/Qwen2.5-VL-3B-Instruct,base_url=http://localhost:30000/v1 \
    --tasks mmmu_val \
    --batch_size 16


branch: main
commit: v0.6-124-g6906172e

| Tasks  |Filter|n-shot| Metric |   |Value |   |Stderr|
|--------|------|-----:|--------|---|-----:|---|------|
|mmmu_val|none  |     0|mmmu_acc|↑  |0.4667|±  |N/A   |


Throughput Summary

|      Metric      |  Value  |  Unit  |
|------------------|--------:|--------|
|total_gen_tokens  |2632.0000|tokens  |
|total_elapsed_time|1938.2665|seconds |
|avg_speed         |   1.3579|tokens/s|
```

## Checklist

- [x] Pre-commit formatting
- [x] Unit tests (shared with Qwen2-VL kernel)
- [x] Accuracy and speed benchmarks provided
- [x] Code style followed

---
### CI States

Latest PR Test (Base): :x: [Run #30969480659](https://github.com/sgl-project/sglang/actions/runs/30969480659)
Latest PR Test (Extra): :x: [Run #30969480520](https://github.com/sgl-project/sglang/actions/runs/30969480520)

### CODE REVIEW

**Comment by flutist** [comment]
@mickqian @yizhang2077 Sorry to bother you, but could you please help me merge this PR file? This solved the problem. If there's anything else I can do, I'll continue. I'm very happy to hear your response.

**Comment by flutist** [comment]
Hi team! This PR extends the Triton vision rotary position embedding kernel (already merged for Qwen2-VL) to Qwen2.5-VL. Only 14 lines added with full accuracy verification.

Would appreciate a review when you get a chance 🙏

@BBuf @Edwardf0t1 @Fridge003 @HaiShaw @Ying1123 @ch-wan @ispobock @merrymercy @mickqian @yizhang2077

**Review by mickqian — APPROVED** [review]

**Comment by flutist** [comment]
could you help to bypassing since CI flaky, thanks   @mickqian

**Comment by AgainstEntropy** [comment]
hi @flutist , can you fix the lint?

**Comment by flutist** [comment]
> hi @flutist , can you fix the lint?

i already fix the lint, thanks for tips.@AgainstEntropy

**Comment by flutist** [comment]
Sorry to bother you, but could you please add the 'run-ci' label to this PR or bypassing since CI flaky? thanks a lot.@AgainstEntropy @mickqian

**Comment by flutist** [comment]
Sorry to bother you, but could you help to bypass since CI flaky? thanks a lot.@AgainstEntropy @mickqian

**Comment by flutist** [comment]
@mickqian @merrymercy @Ying1123 @Fridge003 @ispobock @HaiShaw @ch-wan @BBuf @Edwardf0t1

Sorry to bother you, but could you help to bypass since CI flaky? Thanks a lot. 🙏

**Comment by flutist** [comment]
@ping1jing2 could you help to bypassing since CI flaky, thanks

**Comment by flutist** [comment]
@mickqian @merrymercy @Ying1123 @Fridge003 @ispobock @HaiShaw @ch-wan @BBuf @Edwardf0t1 @ping1jing2 

Sorry to bother you, but could you help to bypass since CI flaky? Thanks a lot. 🙏

**Comment by flutist** [comment]
@mickqian @merrymercy @Ying1123 @Fridge003 @ispobock @HaiShaw @ch-wan @BBuf @Edwardf0t1 @ping1jing2

Sorry to bother you, but could you help to bypass since CI flaky? Thanks a lot. 🙏

File: python/sglang/srt/models/qwen2_5_vl.py [hunk 1]
```diff
@@ -391,6 +391,20 @@ def device(self) -> torch.device:
         return self.patch_embed.proj.weight.device
 
     def rot_pos_emb(self, grid_thw: torch.Tensor) -> torch.Tensor:
+        if _is_cuda and grid_thw.is_cuda:
```
Comment (ping1jing2) [inline python/sglang/srt/models/qwen2_5_vl.py]: please use `srt.platforms.current_platform` instead of `is_cuda`

Comment (flutist) [reply python/sglang/srt/models/qwen2_5_vl.py]: already changed, thanks

**Comment by flutist** [comment]
@ping1jing2 
---

End-to-end test passed after the change.

**Test command:**
```bash
OPENAI_API_KEY=dummy python3 -m lmms_eval \
    --model openai_compatible \
    --model_args model_version=Qwen/Qwen2.5-VL-3B-Instruct,base_url=http://localhost:30000/v1 \
    --tasks mmmu_val \
    --batch_size 16
```

**Result:**
| Tasks  |Filter|n-shot| Metric |   |Value|   |Stderr|
|--------|------|-----:|--------|---|----:|---|------|
|mmmu_val|none  |     0|mmmu_acc|↑  | 0.47|±  |N/A   |

Accuracy is consistent with the baseline — no regression.

**Comment by flutist** [comment]
@mickqian @merrymercy @Ying1123 @Fridge003 @ispobock @HaiShaw @ch-wan @BBuf @Edwardf0t1 @ping1jing2

Sorry to bother you, but could you help to bypass since CI flaky? Thanks a lot. 🙏

**Comment by flutist** [comment]
@mickqian @merrymercy @Ying1123 @Fridge003 @ispobock @HaiShaw @ch-wan @BBuf @Edwardf0t1 @ping1jing2

Sorry to bother you, but could you help to bypass since CI flaky? Thanks a lot. 🙏

**Comment by flutist** [comment]
@mickqian @merrymercy @Ying1123 @Fridge003 @ispobock @HaiShaw @ch-wan @BBuf @Edwardf0t1 @ping1jing2

Sorry to bother you, but could you help to bypass since CI flaky? Thanks a lot. 🙏

**Comment by flutist** [comment]
@mickqian @merrymercy @Ying1123 @Fridge003 @ispobock @HaiShaw @ch-wan @BBuf @Edwardf0t1 @ping1jing2

Sorry to bother you, but could you help to bypass since CI flaky? Thanks a lot. 🙏

**Comment by flutist** [comment]
@mickqian @merrymercy @Ying1123 @Fridge003 @ispobock @HaiShaw @ch-wan @BBuf @Edwardf0t1 @ping1jing2

Sorry to bother you, but could you help to bypass since CI flaky? Thanks a lot. 🙏

**Comment by flutist** [comment]
@mickqian @merrymercy @Ying1123 @Fridge003 @ispobock @HaiShaw @ch-wan @BBuf @Edwardf0t1 @ping1jing2

Sorry to bother you, but could you help to bypass since CI flaky? Thanks a lot. 🙏

**Comment by flutist** [comment]
@mickqian @merrymercy @Ying1123 @Fridge003 @ispobock @HaiShaw @ch-wan @BBuf @Edwardf0t1 @ping1jing2

Sorry to bother you, but could you help to bypass since CI flaky? Thanks a lot. 🙏

**Comment by flutist** [comment]
@mickqian @merrymercy @Ying1123 @Fridge003 @ispobock @HaiShaw @ch-wan @BBuf @Edwardf0t1 @ping1jing2

Sorry to bother you, but could you help to bypass since CI flaky? Thanks a lot. 🙏

**Comment by flutist** [comment]
@mickqian @merrymercy @Ying1123 @Fridge003 @ispobock @HaiShaw @ch-wan @BBuf @Edwardf0t1 @ping1jing2

Sorry to bother you, but could you help to bypass since CI flaky? Thanks a lot. 🙏
